boot-microservices/src/main/java/org/ow2/proactive/boot/microservices/IAMStarter.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.boot.microservices;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.*;
+
+import org.apache.log4j.Logger;
+
+
+public enum IAMStarter {
+
+    //singleton instance of IAMStarter
+    INSTANCE;
+
+    private static final Logger logger = Logger.getLogger(IAMStarter.class);
+
+    private static final String separator = File.separator;
+
+    private static final String os = System.getProperty("os.name");
+
+    private static final String microservice_name = "iam.war";
+
+    private static final String ready_marker = "Ready to process requests";
+
+    private static final int timeout = 180;
+
+    private static Process process;
+
+    private static List<String> command = new ArrayList<String>();
+
+    private static boolean started = false;
+
+    /**
+     *  Start IAM microservice
+     */
+    public Process start(String pa_home, String microservices_path, String[] jvmArgs)
+            throws InterruptedException, IOException, ExecutionException {
+
+        if (!started  && buildJavaCommand(pa_home,jvmArgs) && buildMicroservicePath(pa_home,microservices_path)) {
+
+            System.out.println("Starting IAM microservice");
+
+            ProcessBuilder processBuilder = new ProcessBuilder(command);
+            processBuilder.redirectError(ProcessBuilder.Redirect.INHERIT);
+
+            process = processBuilder.start();
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make the enclosing method "static" or remove this set. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2696)